### PR TITLE
Run presubmit tests for cocoon and engine automatically on luci.

### PR DESCRIPTION
**Pull request by godofredoc**
Both cocoon and engine are ready to start running luci presubmit tests
automatically. This add support for cocoon repo and removes the need for
CQ+1 on engine commits.

https://github.com/flutter/flutter/issues/43348

### CODE REVIEW

File: app_dart/lib/src/request_handlers/github_webhook.dart [hunk 1]
```diff
@@ -130,7 +141,14 @@ class GithubWebhook extends RequestHandler<Body> {
...
     // null indicates unknown. Err on the side of allowing the job to run.
-    if (event.pullRequest.mergeable != false && await _checkForCqLabel(labels)) {
+
+    // We flutter/flutter tests need to be optimized before enforcing CQ.
+    bool runCQ = true;
+    if (needsCQLabelList.contains(event.repository.fullName.toLowerCase())) {
+      runCQ = await _checkForCqLabel(labels);
+    }
+
+    if (event.pullRequest.mergeable != false && runCQ) {
```
Comment (dnfield) [inline app_dart/lib/src/request_handlers/github_webhook.dart]: I think the `mergeable` should probably be removed - it's not trustworthy based on other experience.

Comment (godofredoc) [reply app_dart/lib/src/request_handlers/github_webhook.dart]: Done!

File: app_dart/lib/src/request_handlers/github_webhook.dart [hunk 2]
```diff
@@ -18,6 +18,12 @@ import '../request_handling/exceptions.dart';
 import '../request_handling/request_handler.dart';
 import '../service/buildbucket.dart';
 
+/// List of Github supported repos.
+const List<String> supportedRepos = <String>['engine', 'flutter', 'cocoon'];
+
+/// List of repos that require CQ+1 label.
+const List<String> needsCQLabelList = <String>['flutter/flutter'];
```
Comment (dnfield) [inline app_dart/lib/src/request_handlers/github_webhook.dart]: Both of these should be `Set`s.

**Comment by dnfield** [comment]
fixes https://github.com/flutter/flutter/issues/45562

File: app_dart/lib/src/request_handlers/github_webhook.dart (hunk 2 above)
Comment (godofredoc) [reply app_dart/lib/src/request_handlers/github_webhook.dart]: Done!

**Pull request merged**